src/EnergyPlus/HVACVariableRefrigerantFlow.cc

```diff
@@ -7347,119 +7832,123 @@ namespace HVACVariableRefrigerantFlow {
-                SimDXCoil("",
-                          Off,
-                          FirstHVACIteration,
-                          VRFTU(VRFTUNum).HeatCoilIndex,
-                          OpMode,
-                          PartLoadRatio,
-                          OnOffAirFlowRatio,
-                          _,
-                          MaxHeatingCapacity(VRFCond));
+                SimDXCoil("", Off, FirstHVACIteration, this->HeatCoilIndex, OpMode, PartLoadRatio, OnOffAirFlowRatio, _, MaxHeatingCapacity(VRFCond));
             } else {
-                SimDXCoil("", Off, FirstHVACIteration, VRFTU(VRFTUNum).HeatCoilIndex, OpMode, 0.0, OnOffAirFlowRatio);
+                SimDXCoil("", Off, FirstHVACIteration, this->HeatCoilIndex, OpMode, 0.0, OnOffAirFlowRatio, _);
             }
             LoopDXHeatCoilRTF = LoopDXCoilRTF;
         } else {
             LoopDXHeatCoilRTF = 0.0;
         }
 
         // if draw through, simulate coils then fan
-        if (VRFTU(VRFTUNum).FanPlace == DrawThru) {
-            if (VRFTU(VRFTUNum).fanType_Num == DataHVACGlobals::FanType_SystemModelObject) {
+        if (this->FanPlace == DrawThru) {
+            if (this->fanType_Num == DataHVACGlobals::FanType_SystemModelObject) {
                 if (OnOffAirFlowRatio > 0.0) {
-                    HVACFan::fanObjs[VRFTU(VRFTUNum).FanIndex]->simulate(1.0 / OnOffAirFlowRatio, ZoneCompTurnFansOn, ZoneCompTurnFansOff, _);
+                    HVACFan::fanObjs[this->FanIndex]->simulate(1.0 / OnOffAirFlowRatio, ZoneCompTurnFansOn, ZoneCompTurnFansOff, _);
                 } else {
-                    HVACFan::fanObjs[VRFTU(VRFTUNum).FanIndex]->simulate(PartLoadRatio, ZoneCompTurnFansOn, ZoneCompTurnFansOff, _);
+                    HVACFan::fanObjs[this->FanIndex]->simulate(PartLoadRatio, ZoneCompTurnFansOn, ZoneCompTurnFansOff, _);
                 }
 
             } else {
-                Fans::SimulateFanComponents("", FirstHVACIteration, VRFTU(VRFTUNum).FanIndex, FanSpeedRatio, ZoneCompTurnFansOn, ZoneCompTurnFansOff);
+                Fans::SimulateFanComponents("", FirstHVACIteration, this->FanIndex, FanSpeedRatio, ZoneCompTurnFansOn, ZoneCompTurnFansOff);
             }
         }
 
         // track fan power per terminal unit for calculating COP
-        if (VRFTU(VRFTUNum).fanType_Num == DataHVACGlobals::FanType_SystemModelObject) {
-            VRFTU(VRFTUNum).FanPower = HVACFan::fanObjs[VRFTU(VRFTUNum).FanIndex]->fanPower();
+        if (this->fanType_Num == DataHVACGlobals::FanType_SystemModelObject) {
+            this->FanPower = HVACFan::fanObjs[this->FanIndex]->fanPower();
         } else {
-            VRFTU(VRFTUNum).FanPower = Fans::GetFanPower(VRFTU(VRFTUNum).FanIndex);
+            this->FanPower = Fans::GetFanPower(this->FanIndex);
+        }
+
+        // run supplemental heating coil
+        Real64 SuppPLR = this->SuppHeatPartLoadRatio;
```

Comment:
Unless SuppPLR is used somewhere else you could have put this inside the IF block and only executed this line if a supplemental coil was present.

Reply:
@rraustad You are right. Done.